static/home/js/ligand_structural_search.js

```diff
@@ -0,0 +1,283 @@
+          hash = "#by-id";
+          break;
+        case 'ligand_bulk_search_error_msg':
+          hash = "#bulk-search";
+          break;
+        case 'ligand_structural_search_error_msg':
+          hash = "#sim-sub-search";
+          break;      
+        default:
+          break;
+      }
+
+      setTimeout(() => {
+        location.hash = hash;
+      }, 100);
+    }
+  })
+
+  $("#selection-autocomplete-ligand-by-name")
+  .on('change',function (e) {$("#ligand_bulk_search_by_name_error_msg").text('');});
+
+  $("#selection_ligand_bulk_search_by_names_textarea")
+  .on('change',function (e) {$("#ligand_bulk_search_by_names_error_msg").text('');});
+
+  $("#selection_ligand_bulk_search_by_id_textarea, #selection_ligand_by_id_fields")
+  .on('change',function (e) {$("#ligand_bulk_search_by_id_error_msg").text('');});
+
+  $("#selection_ligand_bulk_search_textarea, #selection_ligand_bulk_search_search_type, #selection_ligand_bulk_search_stereochemistry")
+  .on('change',function (e) {$("#ligand_bulk_search_error_msg").text('');});
+
+  $("#ligand_structural_search_smiles,#ligand_structural_search_search_type,#ligand_structural_search_input_type,#ligand_structural_search_similarity_threshold,#ligand_structural_search_stereochemistry")
+  .on('change',function (e) {$("#ligand_structural_search_error_msg").text('');});
+
+});
+
+function submitSelectionLigandBulkSearchByName(url) {
+
+
+  toggleButtonClass('selection-button-ligand-by-name');
+
+  let search_params_data = {
+    search_text : $("#selection-autocomplete-ligand-by-name").val().trim(),
+    search_type : 'name',
+    // limit : $("#ligand_structural_search_limit").val().trim()
+  }
+
+  // set CSRF csrf_token
+  $.ajaxSetup({
+    headers:
+    { "X-CSRFToken": csrf_token }
+  });
+
+  // Submit search parameters
+  $.post("/ligand/read_input_ligand_bulk_search", search_params_data,  function (data) {
+    setTimeout(function(){
+      toggleButtonClass('selection-button-ligand-by-name');
+      window.location.href = url;
+    }, 500);
+    
+  })
+  .fail(
+    function(jqXHR, textStatus, errorThrown){
+      if (jqXHR.hasOwnProperty('responseJSON')) {
+        if (jqXHR.responseJSON.hasOwnProperty('status_code')) {
+          if (jqXHR.responseJSON.status_code === 400) {
+            if (jqXHR.responseJSON.hasOwnProperty('msg')) {
+              location.hash = '#';
+              location.reload();
+              return;
+            }
+          }
+        }
+          
+      }
+      toggleButtonClass('selection-button-ligand-by-name');
+      showAlert("Something went wrong, please try again or contact us.", "danger");
+    });
+
+
+}
+
+function submitSelectionLigandBulkSearchByNames(url) {
+
+
+  toggleButtonClass('selection-button-ligand-by-names');
+
+  let search_params_data = {
+    search_text : $("#selection_ligand_bulk_search_by_names_textarea").val().trim(),
+    search_type : 'names',
+    // limit : $("#ligand_structural_search_limit").val().trim()
```

Comment:
please clean if not needed

Reply:
Solved in [bb9beb8](https://github.com/protwis/protwis/pull/550/commits/bb9beb8c3231460510e199f66a17a1c512c6c8e0).